server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -216,7 +217,8 @@ private static Version fromIdSlow(int id) {
         } else {
             luceneVersion = versions.get(index).luceneVersion;
         }
-        return new Version(id, luceneVersion);
+        // TODO: assume this is an old version that has transport version == release version
```

Comment:
Is this TODO satisfied? It looks like TransportVersion.fromId uses the id as is?

Reply:
This TODO is more aimed at when we do separate out versions - we may need to do something similar to lucene version, to try and infer the transport version. We'll need to see what uses of this are left when we finish the transportversion refactor